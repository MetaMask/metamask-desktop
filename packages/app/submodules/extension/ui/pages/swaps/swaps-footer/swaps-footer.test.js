import React from 'react';

import { renderWithProvider, fireEvent } from '../../../../test/jest';
import SwapsFooter from '.';

const createProps = (customProps = {}) => {
  return {
    onCancel: jest.fn(),
    onSubmit: jest.fn(),
    submitText: 'submitText',
    disabled: false,
    showTermsOfService: true,
    ...customProps,
  };
};

describe('SwapsFooter', () => {
  it('renders the component with initial props', () => {
    const props = createProps();
    const { container, getByText } = renderWithProvider(
      <SwapsFooter {...props} />,
    );
    expect(getByText(props.submitText)).toBeInTheDocument();
    expect(getByText('Back')).toBeInTheDocument();
    expect(getByText('Terms of service')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('clicks on a block explorer link', () => {
    global.platform = { openTab: jest.fn() };
    const props = createProps();
    const { getByText } = renderWithProvider(<SwapsFooter {...props} />);
    expect(getByText(props.submitText)).toBeInTheDocument();
    fireEvent.click(getByText('Terms of service'));
    expect(global.platform.openTab).toHaveBeenCalledWith({
      url: 'https://metamask.io/terms.html',
    });
  });
});
