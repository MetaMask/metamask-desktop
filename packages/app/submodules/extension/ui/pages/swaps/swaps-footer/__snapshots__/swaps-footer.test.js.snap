// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapsFooter renders the component with initial props 1`] = `
<div>
  <div
    class="swaps-footer"
  >
    <div
      class="swaps-footer__buttons"
    >
      <div
        class="page-container__footer swaps-footer__custom-page-container-footer-class"
      >
        <footer>
          <button
            class="button btn--rounded btn-secondary page-container__footer-button swaps-footer__custom-page-container-footer-button-class"
            data-testid="page-container-footer-cancel"
            role="button"
            tabindex="0"
          >
            Back
          </button>
          <button
            class="button btn--rounded btn-primary page-container__footer-button swaps-footer__custom-page-container-footer-button-class"
            data-testid="page-container-footer-next"
            role="button"
            tabindex="0"
          >
            submitText
          </button>
        </footer>
      </div>
    </div>
    <div
      class="swaps-footer__bottom-text"
    >
      Terms of service
    </div>
  </div>
</div>
`;
